### PR TITLE
feat/[2045]: Api optimization | My timesheet | time/work-summary

### PR DESCRIPTION
**Pull request by yuhanga-rtc**
### Summary

-

### How to test

1.

### Project Checklist

- [ ] Changes build without any errors
- [ ] Have written adequate test cases
- [ ] Done developer testing in
  - [ ] Chrome
  - [ ] Firefox
  - [ ] Safari
- [ ] Code is formatted with `npm run format`
- [ ] Code is linted with `npm run check-lint`
- [ ] No unnecessary comments left in code
- [ ] Made corresponding changes to the documentation

#### Other

- [ ] New atomic components added
- [ ] New molecules added
- [ ] New pages(routes) added
- [ ] New dependencies installed

### PR Checklist

- [ ] Pull request is raised from the correct source branch
- [ ] Pull request is raised to the correct destination branch
- [ ] Pull request is raised with correct title
- [ ] Pull request is self reviewed
- [ ] Pull request is self assigned
- [ ] Suitable pull request status labels are added (`ready-for-code-review`)

### Additional Information
-

### CODE REVIEW

File: backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java [hunk 1]
```diff
@@ -70,10 +70,10 @@ public ResponseEntity<ResponseEntityDto> getActiveTimeSlot() {
 	}
 
 	@Operation(summary = "Work summary", description = "Returns attendance summary of an employee")
-	@PreAuthorize("hasAnyRole('ROLE_SUPER_ADMIN', 'ATTENDANCE_EMPLOYEE')")
-	@GetMapping(value = "/work-summary", produces = MediaType.APPLICATION_JSON_VALUE)
+	@PreAuthorize("hasRole('ROLE_ATTENDANCE_EMPLOYEE')")
```
[the remark this answers is not in the document]
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java:73]: According to our role hierarchy we use least privilege. So no need to specify a role higher than the ATTENDANCE_EMPLOYEE

**Review by ThinuwanW — COMMENTED** [review]
## 🤖 Claude Code Review

This PR optimizes the work-summary endpoint but introduces a few regressions. The most concerning is the validation rewrite: removing `@NotNull` + `@Valid` while relying on `Validation.isInvalidStartAndEndDate`, which treats a null `endDate` as valid and rejects same-day queries. The authorization tightening (dropping SUPER_ADMIN) is also inconsistent with every other endpoint in this controller.

Found **16** new issue(s): 🔴 3 important, 🟡 7 suggestion(s), 🟣 6 nit(s)

File: backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java [hunk 2]
```diff
@@ -279,23 +280,28 @@ public ResponseEntityDto getActiveTimeSlot() {
...
+
 		User currentUser = userService.getCurrentUser();
-		log.info("getEmployeeTotalWorkBreakHours: execution started by user: {}", currentUser.getUserId());
 
-		Optional<Employee> employeeOptional = employeeDao.findById(currentUser.getUserId());
-		if (employeeOptional.isEmpty()) {
-			throw new EntityNotFoundException(
-					messageUtil.getMessage(PeopleMessageConstant.PEOPLE_ERROR_EMPLOYEE_NOT_FOUND));
+		if (Validation.isInvalidStartAndEndDate(employeeAttendanceSummaryFilterDto.getStartDate(),
+				employeeAttendanceSummaryFilterDto.getEndDate())) {
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java:290]: 🤖 **Claude** · 🔴 **Important**

`Validation.isInvalidStartAndEndDate` returns `false` (i.e. valid) when `endDate == null` — see `Validation.java:80-82`. Combined with the removal of `@NotNull` on the DTO fields and `@Valid` on the controller parameter, a request with a missing `endDate` (or both dates missing) now passes validation and then hits `criteriaBuilder.between(..., startDate, null)` in `TimeRecordRepositoryImpl.getEmployeeAttendanceSummary`, which will throw an NPE/IllegalArgumentException and surface as a 500 instead of a 400. Either restore `@Valid` + `@NotNull` on the DTO, or add explicit null checks here (e.g. `if (start == null || end == null) throw new ModuleException(...)`).

File: backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java [hunk 3]
```diff
@@ -279,23 +280,28 @@ public ResponseEntityDto getActiveTimeSlot() {
...
+		log.info("getEmployeeAttendanceSummary: execution started");
+
 		User currentUser = userService.getCurrentUser();
-		log.info("getEmployeeTotalWorkBreakHours: execution started by user: {}", currentUser.getUserId());
 
-		Optional<Employee> employeeOptional = employeeDao.findById(currentUser.getUserId());
-		if (employeeOptional.isEmpty()) {
-			throw new EntityNotFoundException(
-					messageUtil.getMessage(PeopleMessageConstant.PEOPLE_ERROR_EMPLOYEE_NOT_FOUND));
+		if (Validation.isInvalidStartAndEndDate(employeeAttendanceSummaryFilterDto.getStartDate(),
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java]: 🤖 **Claude** · 🔴 **Important**

`Validation.isInvalidStartAndEndDate` rejects the case `startDate == endDate` (it uses `!startDate.isBefore(endDate)`). Previously the endpoint only required non-null dates, so single-day queries worked. The frontend's `TimesheetDailyLogFilter` Custom Range allows the user to pick the same date for start and end (`selectedDates[0] == selectedDates[1]`), so this is a user-facing regression on My Timesheet. Use `startDate.isAfter(endDate)` (matching `getManagerAttendanceSummary` at line 738) or create a helper that allows equal dates.

File: backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java (hunk 1 above)
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java:73]: 🤖 **Claude** · 🔴 **Important**

Authorization was tightened from `hasAnyRole('ROLE_SUPER_ADMIN', 'ATTENDANCE_EMPLOYEE')` to `hasRole('ROLE_ATTENDANCE_EMPLOYEE')`, silently revoking access for super admins. Every other endpoint in this controller (and all related controllers — 175 occurrences across the repo) still grants SUPER_ADMIN access alongside the module role. If intentional, document why; otherwise restore `hasAnyRole('ROLE_SUPER_ADMIN', 'ATTENDANCE_EMPLOYEE')` for consistency.

File: backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java [hunk 4]
```diff
@@ -70,10 +70,10 @@ public ResponseEntity<ResponseEntityDto> getActiveTimeSlot() {
...
+	@GetMapping(value = "/work-summary")
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java:74]: 🤖 **Claude** · 🟡 **Suggestion**

`produces = MediaType.APPLICATION_JSON_VALUE` was dropped from this `@GetMapping`. Functionally minor (the `@RestController` still serializes to JSON via `HttpMessageConverter`), but every other endpoint in this controller declares it. Keep it for consistency and to ensure proper content negotiation on `Accept` headers.

File: backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java [hunk 5]
```diff
@@ -70,10 +70,10 @@ public ResponseEntity<ResponseEntityDto> getActiveTimeSlot() {
...
 	public ResponseEntity<ResponseEntityDto> getEmployeeAttendanceSummary(
-			@Valid EmployeeAttendanceSummaryFilterDto employeeAttendanceSummaryFilterDto) {
+			EmployeeAttendanceSummaryFilterDto employeeAttendanceSummaryFilterDto) {
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java:76]: 🤖 **Claude** · 🟡 **Suggestion**

`@Valid` was removed from the DTO parameter. Even though `@NotNull` is gone today, dropping `@Valid` makes any future bean-validation annotation (e.g. `@PastOrPresent`, `@NotBlank`) silently no-op. Suggest keeping `@Valid` even if the DTO currently has no constraints, matching every other endpoint in this controller.

File: backend/src/main/java/com/skapp/community/timeplanner/service/TimeService.java [hunk 6]
```diff
@@ -34,6 +34,7 @@ public interface TimeService {
 
 	ResponseEntityDto getActiveTimeSlot();
 
+	@Transactional
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/timeplanner/service/TimeService.java]: 🤖 **Claude** · 🟡 **Suggestion**

`@Transactional` placed on the interface defaults to read-write (propagation=REQUIRED, readOnly=false), but the implementation overrides with `@Transactional(readOnly = true)`. The impl wins for the proxy that's invoked, but the mismatch is misleading and fragile (e.g. a subclass that overrides this method would silently inherit a writable transaction). Either drop the interface annotation (Spring recommends placing `@Transactional` only on concrete classes) or change it to `@Transactional(readOnly = true)` to match the impl.

File: backend/src/main/java/com/skapp/community/timeplanner/repository/TimeRecordRepositoryImpl.java [hunk 7]
```diff
@@ -67,17 +67,14 @@ public AttendanceSummaryDto getEmployeeAttendanceSummary(List<Long> employeeIds,
...
-		Predicate[] predArray = new Predicate[predicates.size()];
-		predicates.toArray(predArray);
-		criteriaQuery.where(predArray);
+		criteriaQuery.where(predicates.toArray(new Predicate[0]));
 		criteriaQuery.select(criteriaBuilder.construct(AttendanceSummaryDto.class,
-				criteriaBuilder.coalesce(criteriaBuilder.sum(root.get(TimeRecord_.workedHours)), 0.0),
-				criteriaBuilder.coalesce(criteriaBuilder.sum(root.get(TimeRecord_.breakHours)), 0.0)));
-		TypedQuery<AttendanceSummaryDto> typedQuery = entityManager.createQuery(criteriaQuery);
-		return typedQuery.getSingleResult();
+				criteriaBuilder.coalesce(criteriaBuilder.sum(root.get(TimeRecord_.workedHours)), 0.0f),
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/timeplanner/repository/TimeRecordRepositoryImpl.java:75]: 🤖 **Claude** · 🟣 **Nit**

Good change — `0.0f` is type-correct for the Float `workedHours`/`breakHours` and the `AttendanceSummaryDto(Float, Float)` constructor. While here, consider the same fix in `findManagerAssignUsersAttendanceSummary` (lines 124-125) and `findTimeSheetSummaryData` (lines 154-157), which still pass `0.0` (Double) into the same `coalesce` over `Float` fields.

File: backend/src/main/java/com/skapp/community/timeplanner/payload/request/EmployeeAttendanceSummaryFilterDto.java [hunk 8]
```diff
@@ -10,10 +9,8 @@
 @Setter
 public class EmployeeAttendanceSummaryFilterDto {
 
-	@NotNull
 	private LocalDate startDate;
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/timeplanner/payload/request/EmployeeAttendanceSummaryFilterDto.java:12]: 🤖 **Claude** · 🟡 **Suggestion**

Removing both `@NotNull` annotations leaves this DTO with no bean-validation safety net. Sibling DTO `ManagerAttendanceSummaryFilterDto` still uses `@NotNull` on `startDate`/`endDate` — keep the two consistent. Either restore `@NotNull` here (and re-add `@Valid` in the controller), or accept that all null-handling must happen explicitly in the service (see the related finding on `Validation.isInvalidStartAndEndDate` returning valid for null endDate).

File: backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java [hunk 9]
```diff
@@ -279,23 +280,28 @@ public ResponseEntityDto getActiveTimeSlot() {
 	}
 
 	@Override
+	@Transactional(readOnly = true)
 	public ResponseEntityDto getEmployeeAttendanceSummary(
 			EmployeeAttendanceSummaryFilterDto employeeAttendanceSummaryFilterDto) {
+		log.info("getEmployeeAttendanceSummary: execution started");
+
 		User currentUser = userService.getCurrentUser();
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java]: 🤖 **Claude** · 🟣 **Nit**

`User currentUser = userService.getCurrentUser();` is fetched before the date validation. If the request is malformed, you still incur the user lookup. Minor — move it after the validation block so invalid requests fail fast.

Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java]: 🤖 **Claude** · 🟡 **Suggestion**

Behavioral regression: the original method called `employeeDao.findById(currentUser.getUserId())` and threw `EntityNotFoundException` (PEOPLE_ERROR_EMPLOYEE_NOT_FOUND) when no Employee existed. That check is gone. If a current user with the ROLE_ATTENDANCE_EMPLOYEE authority lacks a matching Employee row (data drift, misconfigured user, etc.), the repo query simply returns zero rows and `coalesce` yields `0.0f`, so the API silently responds with "0h 0m" instead of a meaningful 4xx error. Restore the existence check (or document explicitly why it's safe to drop it) so downstream consumers can distinguish "no data" from "user is not an employee".

File: backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java [hunk 10]
```diff
@@ -279,23 +280,28 @@ public ResponseEntityDto getActiveTimeSlot() {
 	}
 
 	@Override
+	@Transactional(readOnly = true)
 	public ResponseEntityDto getEmployeeAttendanceSummary(
 			EmployeeAttendanceSummaryFilterDto employeeAttendanceSummaryFilterDto) {
+		log.info("getEmployeeAttendanceSummary: execution started");
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java:286]: 🤖 **Claude** · 🟡 **Suggestion**

Observability regression: the previous log line was `log.info("getEmployeeTotalWorkBreakHours: execution started by user: {}", currentUser.getUserId())` and the new one drops the user identifier entirely. Every other method in this file (e.g. lines 242, 277, 309, 312, 343) keeps the userId in its info log for traceability. Include `currentUser.getUserId()` (and ideally the date range) so production traces remain useful. The same applies to line 303 — `"execution ended"` without context is noise.

File: backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java [hunk 11]
```diff
@@ -279,23 +280,28 @@ public ResponseEntityDto getActiveTimeSlot() {
...
 		User currentUser = userService.getCurrentUser();
-		log.info("getEmployeeTotalWorkBreakHours: execution started by user: {}", currentUser.getUserId());
 
-		Optional<Employee> employeeOptional = employeeDao.findById(currentUser.getUserId());
-		if (employeeOptional.isEmpty()) {
-			throw new EntityNotFoundException(
-					messageUtil.getMessage(PeopleMessageConstant.PEOPLE_ERROR_EMPLOYEE_NOT_FOUND));
+		if (Validation.isInvalidStartAndEndDate(employeeAttendanceSummaryFilterDto.getStartDate(),
+				employeeAttendanceSummaryFilterDto.getEndDate())) {
+			throw new ModuleException(TimeMessageConstant.TIME_ERROR_START_DATE_END_DATE_NOT_VALID);
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java:291]: 🤖 **Claude** · 🟡 **Suggestion**

`TimeMessageConstant.TIME_ERROR_START_DATE_END_DATE_NOT_VALID` resolves to the message key `api.error.time.trend.start.end.date.not.valid` (TimeMessageConstant.java:34) — the "trend" segment was added for the clock-in/out trend chart endpoint, not the work-summary feature. Reusing it here will surface a misleading localized error to clients of `/work-summary`. Add a dedicated key (e.g. `api.error.time.work-summary.invalid-date-range`) or rename the existing key to be generic.

File: backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java (hunk 1 above)
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java:73]: 🤖 **Claude** · 🟣 **Nit**

Convention drift: this single endpoint uses `hasRole('ROLE_ATTENDANCE_EMPLOYEE')` while every other `@PreAuthorize` in this controller writes the role without the `ROLE_` prefix (Spring auto-prepends it). The two forms are functionally equivalent here, but the inconsistency stands out on review. Either change this back to `hasAnyRole('ROLE_SUPER_ADMIN', 'ATTENDANCE_EMPLOYEE')` (also restoring SUPER_ADMIN access flagged in the first pass) or pick one prefix convention and apply it across the file.

File: backend/src/main/java/com/skapp/community/timeplanner/repository/TimeRecordRepositoryImpl.java [hunk 12]
```diff
@@ -67,17 +67,14 @@ public AttendanceSummaryDto getEmployeeAttendanceSummary(List<Long> employeeIds,
...
+				criteriaBuilder.coalesce(criteriaBuilder.sum(root.get(TimeRecord_.breakHours)), 0.0f)));
+		return entityManager.createQuery(criteriaQuery).getSingleResult();
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/timeplanner/repository/TimeRecordRepositoryImpl.java:77]: 🤖 **Claude** · 🟣 **Nit**

`getSingleResult()` here is not wrapped in a try/catch, but the sibling method `findManagerAssignUsersAttendanceSummary` (same file, lines 129-134) catches `NoResultException` and returns `new AttendanceSummaryDto(0.0F, 0.0F)`. With `coalesce(sum, 0.0f)` the no-result branch is unreachable in practice, so either remove the try/catch from the sibling or add an equivalent one here — pick one defensive style across the file.

File: backend/src/main/java/com/skapp/community/timeplanner/payload/request/EmployeeAttendanceSummaryFilterDto.java (hunk 8 above)
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/timeplanner/payload/request/EmployeeAttendanceSummaryFilterDto.java:12]: 🤖 **Claude** · 🟣 **Nit**

The sibling DTO `ManagerAttendanceSummaryFilterDto` documents each field with `@Schema(description = ...)` for OpenAPI/Swagger. This DTO has neither bean-validation annotations (after this PR) nor `@Schema` descriptions, so the generated API docs for `/work-summary` will list `startDate`/`endDate` as undocumented query parameters. Add `@Schema(description = ...)` to keep docs uniform across the two attendance-summary DTOs.

File: backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java [hunk 13]
```diff
@@ -13,6 +13,7 @@
 import com.skapp.community.common.util.CommonModuleUtils;
 import com.skapp.community.common.util.DateTimeUtils;
 import com.skapp.community.common.util.MessageUtil;
+import com.skapp.community.common.util.Validation;
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java:16]: 🤖 **Claude** · 🟣 **Nit**

📁 **File-level observation**

No tests were added or updated for `getEmployeeAttendanceSummary` despite the behavior changes (date-range validation added, employee-existence check removed, transaction changed to read-only, repo coalesce literal changed). The method has no existing unit/integration coverage in the repo. Add at least one test covering the new validation branch (null dates, equal dates, start-after-end) and the happy path so future refactors don't silently regress.

File: backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java [hunk 14]
```diff
@@ -279,23 +280,28 @@ public ResponseEntityDto getActiveTimeSlot() {
...
-			throw new EntityNotFoundException(
-					messageUtil.getMessage(PeopleMessageConstant.PEOPLE_ERROR_EMPLOYEE_NOT_FOUND));
+		if (Validation.isInvalidStartAndEndDate(employeeAttendanceSummaryFilterDto.getStartDate(),
+				employeeAttendanceSummaryFilterDto.getEndDate())) {
+			throw new ModuleException(TimeMessageConstant.TIME_ERROR_START_DATE_END_DATE_NOT_VALID);
 		}
+
 		AttendanceSummaryDto attendanceSummaryDto = timeRecordDao.getEmployeeAttendanceSummary(
 				List.of(currentUser.getUserId()), employeeAttendanceSummaryFilterDto.getStartDate(),
 				employeeAttendanceSummaryFilterDto.getEndDate());
```
[the remark this answers is not in the document]
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java]: Added a new validation function which met all this requirenments

File: backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java (hunk 2 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java:290]: Added a new validation function which met all this requirements

File: backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java (hunk 3 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java]: Include in new validator

File: backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java (hunk 1 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java:73]: According to our role hierarchy we use least privilege. So no need to specify a role higher than the ATTENDANCE_EMPLOYEE

File: backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java (hunk 4 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java:74]: We do not use it now

File: backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java (hunk 5 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java:76]: We do not use it now and created a new validation function

File: backend/src/main/java/com/skapp/community/timeplanner/service/TimeService.java (hunk 6 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/service/TimeService.java]: removed

File: backend/src/main/java/com/skapp/community/timeplanner/repository/TimeRecordRepositoryImpl.java (hunk 7 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/repository/TimeRecordRepositoryImpl.java:75]: Will do it in the next improvement PR

File: backend/src/main/java/com/skapp/community/timeplanner/payload/request/EmployeeAttendanceSummaryFilterDto.java (hunk 8 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/payload/request/EmployeeAttendanceSummaryFilterDto.java:12]: Now we do not use `@NotNull` . Will remove remaining in the next improvement PRs

File: backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java (hunk 9 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java]: Done

Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java]: No need to validate it. It makes two DB calls which is not necessary. The data fetch call will fail if no employee.

File: backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java (hunk 10 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java:286]: Now we do not log variables. Will remove remaining in the next improvement PRs

File: backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java (hunk 11 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java:291]: Created a new validation function

File: backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java (hunk 1 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/controller/v1/TimeController.java:73]: Now we do not use roles without 'ROLE_'prefix. Will fix remaining in the next improvement PRs

File: backend/src/main/java/com/skapp/community/timeplanner/repository/TimeRecordRepositoryImpl.java (hunk 12 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/repository/TimeRecordRepositoryImpl.java:77]: Need to remove there. We do not use try catch unless it is mandatory.

File: backend/src/main/java/com/skapp/community/timeplanner/payload/request/EmployeeAttendanceSummaryFilterDto.java (hunk 8 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/payload/request/EmployeeAttendanceSummaryFilterDto.java:12]: Will remove remaining in the next improvement PRs

File: backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java (hunk 13 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/timeplanner/service/impl/TimeServiceImpl.java:16]: Tests will be added and updated according to improvement at end
